openwisp_controller/config/base/config.py

```diff
@@ -461,6 +461,34 @@ def get_vpn_context(self):
                 )
         return c
 
+    def get_subnet_division_context(self):
+        # NOTE: Use regex to know which subnet division variables
+        # are used in this config and only provide those contexts
+
+        context = {}
+        qs = self.subnetdivisionindex_set.values(
+            'keyword', 'subnet__subnet', 'ip__ip_address'
+        )
+        for entry in qs:
+            if entry['subnet__subnet'] == entry['ip__ip_address']:
+                context[entry['keyword']] = entry['rule__size']
+            elif entry['ip__ip_address'] is None:
+                context[entry['keyword']] = str(entry['subnet__subnet'])
+            else:
+                context[entry['keyword']] = str(entry['ip__ip_address'])
+
+        # TODO: Find a way to avoid this query
```

Comment:
is it really avoidable?

Reply:
I added this TODO while building a prototype just to remind me to look for possibility to avoid this query. I put some thought to it and I couldn't find a way to avoid it since we are querying `SubnetDivisionIndex` using `config_id` here. 

Since we will not query the prefixlen variable (e.g. `OW_prefixlen`), I think we can remove this code too and add it back in future if needed. 

https://github.com/openwisp/openwisp-controller/blob/8702de6d89f3423ed0c13c2dd260de928f2fb427/openwisp_controller/subnet_division/handlers.py#L68-L71